Implement a method to schedule enqueue message on service bus

charlesamoreira opened the pull request:
This implementation just use method ScheduleMessageAsync from Microsoft.Azure.ServiceBus.Core.MessageSender.
Resolves #155

bruno-brant commented:
Shouldn't be approved before #145.

guilhermeluizsp commented:
@bruno-brant I'm waiting for you to rebase with the master branch since #145 is approved and merged.

bruno-brant commented:
> @bruno-brant I'm waiting for you to rebase with the master branch since #145 is approved and merged.

Done

charlesamoreira committed:
Implement a method to schedule enqueue message on service bus

charlesamoreira committed:
DCO Remediation Commit for Charles de Almeida Moreira <charles.moreira@gmail.com>

I, Charles de Almeida Moreira <charles.moreira@gmail.com>, hereby add my Signed-off-by to this commit: cb1db7923582fa1c34225243812c327f3d5bf114

Signed-off-by: Charles de Almeida Moreira <charles.moreira@gmail.com>

lucianareginalino commented on src/Liquid.Messaging.RabbitMq/RabbitMqProducer.cs:
```diff
@@ -36,6 +36,12 @@ public RabbitMqProducer(IRabbitMqFactory factory, RabbitMqProducerSettings setti
         }
 
+        /// <inheritdoc/>
+        public Task ScheduleMessageAsync(DateTimeOffset scheduleEnqueueTimeUtc, TEntity messageBody, IDictionary<string, object> customProperties = null)
```
method must be implemented.

lucianareginalino commented on src/Liquid.Messaging/Interfaces/ILiquidProducer.cs:
```diff
@@ -22,5 +23,14 @@ public interface ILiquidProducer<TEntity>
+        /// <param name="scheduleEnqueueTimeUtc">Scheduled Time to Enqueue a message.</param>
+        /// <param name="messageBody">Body of message to be sent.</param>
+        /// <param name="customProperties">Message header properties.</param>
+        Task ScheduleMessageAsync(DateTimeOffset scheduleEnqueueTimeUtc, TEntity messageBody, IDictionary<string, object> customProperties = null);
```
this class is the base interface for all adapters (cartridges) implementations, and it's methods must be common features of principal messaging vendors (kafka, rabbit, event hub, etc). Please check if this is a common feature, and if it's not, I sugest you implement a new interface for , using some kind of interface composure.

lucianareginalino commented:
We need more details of this feature: why this is a feature we must have in our base implementations and what is the impact for our users? it's a common features of principal messaging resource vendors (like kafka, rabbit, event hub, etc) and can be implemented on all existing cartridges? Can be just an extension of ILiquidProducer and/or ServiceBusProducer?
Please give us more details of the proposed solution in the issue, and consider these points as well as the review comments in your implementation.
